tests/test_dfa.py

```diff
@@ -1452,4 +1452,230 @@ def test_dfa_repr(self):
         )
         self.assertEqual(
             repr(dfa),
-            "DFA(states={'q0'}, input_symbols={'a'}, transitions={'q0': {'a': 'q0'}}, initial_state='q0', final_states={'q0'}, allow_partial=False)")  # noqa: E501
+            "DFA(states={'q0'}, input_symbols={'a'}, transitions={'q0': {'a': 'q0'}}, "
+            "initial_state='q0', final_states={'q0'}, allow_partial=False)")
+
+    def test_iter_finite(self):
+        """
+        Test that DFA for finite language generates all words
+        """
+        language = {'aa', 'aaa', 'aaba', 'aabbb', 'abaa', 'ababb', 'abbab',
+                    'baa', 'babb', 'bbaa', 'bbabb', 'bbbab'}
+        dfa = DFA.from_finite_language(language, {'a', 'b'})
+        generated_set = {word for word in dfa}
+        self.assertEqual(generated_set, language)
+
+    def test_iter_infinite(self):
+        """
+        Test that language that avoids the pattern '11' generates the correct values in correct order
+        """
+        A = DFA(
+            states={'p0', 'p1', 'p2'},
+            input_symbols={'0', '1'},
+            transitions={
+                'p0': {'0': 'p0', '1': 'p1'},
+                'p1': {'0': 'p0', '1': 'p2'},
+                'p2': {'0': 'p2', '1': 'p2'}
+            },
+            initial_state='p0',
+            final_states={'p0', 'p1'}
+        )
+
+        generator = iter(A)
+        expected = ['',
+                    '0', '1',
+                    '00', '01', '10',
+                    '000', '001', '010', '100', '101',
+                    '0000', '0001', '0010', '0100', '0101', '1000', '1001', '1010']
+        generated_list = [next(generator) for _ in expected]
```

Comment:
@Tagl Couldn't this be rewritten as:

```python
generated_list = list(A)
```

I think that would be more consistent with how the user will actually consume the iterable.

Reply:
No, it can't be done here because the iteration is infinite so I only iterate as many times as there are items in `expected`.

Reply:
@Tagl Ah yes, I'm not used to working with potentially-infinite generators 😅. Makes sense.